src/raw/bson.rs

```diff
@@ -239,6 +245,307 @@ impl<'a> RawBson<'a> {
     }
 }
 
+/// A visitor used to deserialize types backed by raw BSON.
+pub(crate) struct RawBsonVisitor;
+
+impl<'de> Visitor<'de> for RawBsonVisitor {
+    type Value = RawBson<'de>;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(formatter, "a raw BSON reference")
+    }
+
+    fn visit_borrowed_str<E>(self, v: &'de str) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(RawBson::String(v))
+    }
+
+    fn visit_borrowed_bytes<E>(self, bytes: &'de [u8]) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(RawBson::Binary(RawBinary {
+            bytes,
+            subtype: BinarySubtype::Generic,
+        }))
+    }
+
+    fn visit_i8<E>(self, v: i8) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(RawBson::Int32(v.into()))
+    }
+
+    fn visit_i16<E>(self, v: i16) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(RawBson::Int32(v.into()))
+    }
+
+    fn visit_i32<E>(self, v: i32) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(RawBson::Int32(v))
+    }
+
+    fn visit_i64<E>(self, v: i64) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(RawBson::Int64(v))
+    }
+
+    fn visit_u8<E>(self, value: u8) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        crate::de::convert_unsigned_to_signed_raw(value.into())
+    }
+
+    fn visit_u16<E>(self, value: u16) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        crate::de::convert_unsigned_to_signed_raw(value.into())
+    }
+
+    fn visit_u32<E>(self, value: u32) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        crate::de::convert_unsigned_to_signed_raw(value.into())
+    }
+
+    fn visit_u64<E>(self, value: u64) -> std::result::Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        crate::de::convert_unsigned_to_signed_raw(value)
```

Comment:
Nit: might be worth just `use`ing `convert_unsigned_to_signed_raw`?

Reply:
fixed